src/Tizen.NUI/src/internal/BackKeyManager.cs

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright(c) 2019 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+using System;
+using System.ComponentModel;
+using System.Collections.Generic;
+using Tizen.NUI.BaseComponents;
+
+namespace Tizen.NUI
+{
+    /// <summary>
+    /// BackKeyManager is a class to manager back key event.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class BackKeyManager
+    {
+
+        /// <summary>
+        /// BackKeyManager construct.
+        /// </summary>
+        private BackKeyManager()
+        {
+            NUIApplication.GetDefaultWindow().KeyEvent += OnWindowKeyEvent;
+        }
+
+        /// <summary>
+        /// Subscriber.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public List<View> Subscriber{get; set;} = new List<View>();
+
+        /// <summary>
+        /// BackKeyManager static instance.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static BackKeyManager Instance { get; } = new BackKeyManager();
+
+        private Container FindParent(Container child, int step)
+        {
+            Container result = child;
+
+            for (int i = 0; i < step; i++)
+            {
+                result = result?.GetParent();
+            }
+
+            return result;
+        }
+
+        // If false, comparison1 is on top and if true, comparison2 is on top.
+        private bool CompareElevation(Container comparison1, Container comparison2)
+        {
+            bool result = false;
+
+            Container parent1 = comparison1?.GetParent();
+            Container parent2 = comparison2?.GetParent();
+
+            if((parent1 == null && parent2 != null) || (parent2 == null && parent1 != null))
+            {
+                // One is in DefaultLayer but the other is not
+                if(parent2 == null)
+                {
+                    // parent1 is in DefaultLayer.
+                    Layer comparison2AsLayer = comparison2 as Layer;
```

Comment:
is there any possibility that comparison2AsLayer is null?

Reply:
In this case, we have problem that relation between DefaultLayer and other Layer.
They looks like siblings but in fact, they are parent-child.

Because of this situation, when using new Layer, the depth of view can be differ like below.
DefaultLayer - View1 - View2 -> View2.HierarchyDepth is 2
Other Layer - View1 - View2 -> View2.HierarchyDepth is 3.

So, this statement just for one of comparison is in other Layer and the comparison2 is always Layer never null.

Reply:
OK, I see. 
I didn't know that Layer's depth issue... I think we need to fix that in nui side anyway...